indexer/drawing_rule_def.hpp

```diff
@@ -20,7 +20,7 @@ int32_t constexpr kOverlaysMaxPriority = 10000;
   public:
     int m_scale = -1;
     int m_type = -1;
-    int m_index = -1;
+    size_t m_index = std::numeric_limits<size_t>::max(); // an index to RulesHolder.m_dRules[]
```

Comment:
I think, def initialization is useless here, can remove.

Reply:
In many places Key is created via a default constructor, which doesn't assign anything.
So theoretically its possible it'll be passed undefined into e.g. RulesHolder::Find(). Am I right?

Reply:
Yes, no problem here. Anyway, max() is the same as 0 is the same as nothing.
def ctor needed for buffer_vector